server/src/test/java/io/spine/examples/chatspn/server/message/MessageEditingTest.java

```diff
@@ -0,0 +1,228 @@
+    }
+
+    @Test
+    @DisplayName("update a `Message` to the expected state")
+    void state() {
+        Message editedMessage = setUpChatWithEditedMessage(context());
+
+        context().assertState(editedMessage.getId(), Message.class)
+                 .comparingExpectedFieldsOnly()
+                 .isEqualTo(editedMessage);
+    }
+
+    @Test
+    @DisplayName("reject with the `MessageCannotBeEdited` if the message editor is not the chat member")
+    void messageCannotBeEditedRejection() {
+        Chat chat = createRandomChat(context());
+        Message message = sendMessage(chat.getId(),
+                                      chat.getMember(0),
+                                      context());
+        Message wrongUserMessage = message.toBuilder()
+                                          .setUser(GivenUserId.generated())
+                                          .buildPartial();
+        Message editedMessage = editMessage(wrongUserMessage, context());
+        MessageCannotBeEdited expectedRejection = MessageCannotBeEdited
+                .newBuilder()
+                .setId(editedMessage.getId())
+                .setChat(editedMessage.getChat())
+                .setUser(editedMessage.getUser())
+                .setContent(editedMessage.getContent())
+                .vBuild();
+
+        context().assertEvents()
+                 .withType(MessageCannotBeEdited.class)
+                 .message(0)
+                 .isEqualTo(expectedRejection);
+    }
+
+    @Test
+    @DisplayName("emit a `MessageNotEdited` event if message content cannot be edited and archive itself")
+    void messageNotEditedEvent() {
+        Chat chat = createRandomChat(context());
+        Message message = Message
+                .newBuilder()
+                .setId(MessageId.generate())
+                .setUser(chat.getMember(0))
+                .setChat(chat.getId())
+                .setContent(randomString())
+                .buildPartial();
+        Message editedMessage = editMessage(message, context());
+        MessageNotEdited expectedEvent = MessageNotEdited
+                .newBuilder()
+                .setId(editedMessage.getId())
+                .setChat(editedMessage.getChat())
+                .setUser(editedMessage.getUser())
+                .setContent(editedMessage.getContent())
+                .vBuild();
+
+        context().assertEvents()
+                 .withType(MessageNotEdited.class)
+                 .hasSize(1);
+        context().assertEvents()
+                 .withType(MessageNotEdited.class)
+                 .message(0)
+                 .isEqualTo(expectedEvent);
+        context().assertEntity(editedMessage.getId(), MessageEditingProcess.class)
+                 .archivedFlag()
+                 .isTrue();
+    }
+
+    @Nested
+    @DisplayName("lead `MessageAggregate` to")
+    class MessageAggregate {
+
+        @Test
+        @DisplayName("emit a `MessageContentEdited` event")
```

Comment:
`@DisplayName`s, when combined, should turn into valid and meaningful sentences.
```
... lead `MessageAggregate` to emit a `MessageContentEdited` event
```
... does not sound too English. Maybe you meant "... to emission of ..."?